## CI failed — REJECT (compile error under tray feature)

**HEAD**: e1a80a5
**Failing job**: Clippy (tray feature), all 3 platforms
**Errors**:
```
error[E0560]: struct `layout::SplitBorderHit` has no field named `pane_id`
error[E0560]: struct `layout::SplitBorderHit` has no field named `pos`
```

**Root cause**: drag test fixtures reference non-existent `SplitBorderHit.pane_id` / `pos`. Actual struct fields are `split_area: (u16, u16, u16, u16)` and `dir: SplitDir`.

**dev2-1 prior VERIFIED verdict**: stands for non-CI-related dimensions (test scope, logic-vs-glue gate, drag coverage all PASS), but did not catch tray-feature compile error.

**7th silent-deviation pattern**: local `cargo test ✓` claim without `--features=tray` exercise.

dev2 dispatched for r3 fix — fixture correction + tray-feature local clippy now mandatory pre-push.

--- a/src/app/mouse.rs
+++ b/src/app/mouse.rs
@@ -293,6 +293,7 @@ fn handle_up(
     }
 }
 
+#[derive(Debug, PartialEq)]
 enum TabBarClick {
     Tab(usize),
     NewTab,
@@ -621,5 +622,111 @@ mod tests {
         handle_drag(event, &mut layout, &mut state);
         assert_eq!(layout.tabs[0].drag_target, Some(2));
         assert!(layout.tabs[0].drag_target_tab.is_none());
+    }
+
+    // --- Sprint 41 T-3: tab_bar_hit_test coverage ---
+
+    #[test]
+    fn tab_bar_hit_test_first_tab() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("tab0".to_string(), leaf(1, "a")));
+        // First tab starts at col 0: "*" (1) + " tab0 " (6) = 7 cols
+        let result = tab_bar_hit_test(&layout, 0);
+        assert_eq!(result, Some(TabBarClick::Tab(0)));
+    }
+
+    #[test]
+    fn tab_bar_hit_test_second_tab() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("a".to_string(), leaf(1, "x")));
+        layout.add_tab(Tab::new("b".to_string(), leaf(2, "y")));
+        // Tab "a": "*" + " a " = 4 cols (0..3). Separator: 1 col (4).
+        // Tab "b": "*" + " b " = 4 cols (5..8).
+        let result = tab_bar_hit_test(&layout, 5);
+        assert_eq!(result, Some(TabBarClick::Tab(1)));
+    }
+
+    #[test]
+    fn tab_bar_hit_test_new_tab_button() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("x".to_string(), leaf(1, "a")));
+        // Tab "x": 4 cols (0..3). Then " [+] " at col 4..8.
+        let result = tab_bar_hit_test(&layout, 4);
+        assert_eq!(result, Some(TabBarClick::NewTab));
+    }
+
+    #[test]
+    fn tab_bar_hit_test_beyond_all_tabs_returns_none() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("t".to_string(), leaf(1, "a")));
+        // Way past all tabs + [+] button
+        let result = tab_bar_hit_test(&layout, 100);
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn tab_bar_hit_test_empty_layout_returns_new_tab() {
+        let layout = Layout::new();
+        // No tabs → only [+] button at col 0
+        let result = tab_bar_hit_test(&layout, 0);
+        assert_eq!(result, Some(TabBarClick::NewTab));
+    }
+
+    // --- Sprint 41 T-3 r2: drag state machine tests ---
+
+    #[test]
+    fn drag_start_sets_dragging_pane_on_title_hit() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("src".to_string(), leaf(1, "a")));
+        layout.tabs[0].split_focused(SplitDir::Vertical, leaf(2, "b"));
+        layout.active = 0;
+        layout.tabs[0].focus_id = 2;
+        layout.tabs[0].dragging_pane = Some(2);
+        layout.tabs[0].drag_target = None;
+
+        assert_eq!(layout.tabs[0].dragging_pane, Some(2));
+        assert_eq!(layout.tabs[0].drag_target, None);
+    }
+
+    #[test]
+    fn drag_end_clears_dragging_pane() {
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("src".to_string(), leaf(1, "a")));
+        layout.tabs[0].split_focused(SplitDir::Vertical, leaf(2, "b"));
+        layout.active = 0;
+        layout.tabs[0].focus_id = 2;
+        layout.tabs[0].dragging_pane = Some(2);
+        layout.tabs[0].drag_target = Some(1);
+
+        let mut state = MouseState::default();
+        let mut out = MouseOutcome::default();
+        handle_up(up_event(), &mut layout, &mut state, &mut out);
+
+        assert_eq!(
+            layout.tabs[0].dragging_pane, None,
+            "drag-end must clear dragging_pane"
+        );
+    }
+
+    #[test]
+    fn border_drag_state_cleared_on_mouse_up() {
+        let mut state = MouseState {
+            border_drag: Some((
+                crate::layout::SplitBorderHit {
+                    split_area: (0, 1, 60, 38),
+                    dir: SplitDir::Vertical,
+                },
+                ratatui::layout::Rect::new(0, 1, 120, 38),
+            )),
+        };
+        let mut layout = Layout::new();
+        layout.add_tab(Tab::new("t".to_string(), leaf(1, "a")));
+        let mut out = MouseOutcome::default();
+        handle_up(up_event(), &mut layout, &mut state, &mut out);
+
+        assert!(
+            state.border_drag.is_none(),
+            "border_drag must be cleared on mouse-up"
+        );
     }
 }

--- a/src/render.rs
+++ b/src/render.rs
@@ -2310,4 +2310,52 @@ mod tests {
         assert_eq!(transient_state_badge(AgentState::Hang), None);
         assert_eq!(transient_state_badge(AgentState::PermissionPrompt), None);
     }
+
+    // --- Sprint 41 T-3: render logic function coverage ---
+
+    #[test]
+    fn state_color_returns_distinct_colors_for_key_states() {
+        // Pin that key states have distinct colors (logic outcome, not pixel)
+        let idle = state_color(AgentState::Idle);
+        let thinking = state_color(AgentState::Thinking);
+        let tool_use = state_color(AgentState::ToolUse);
+        let ready = state_color(AgentState::Ready);
+        assert_ne!(idle, thinking, "idle vs thinking must differ");
+        assert_ne!(thinking, tool_use, "thinking vs tool_use must differ");
+        assert_ne!(idle, ready, "idle vs ready must differ");
+    }
+
+    #[test]
+    fn state_color_error_states_are_red() {
+        use ratatui::style::Color;
+        assert_eq!(state_color(AgentState::Crashed), Color::Red);
+        assert_eq!(state_color(AgentState::Restarting), Color::Red);
+    }
+
+    #[test]
+    fn highest_priority_state_returns_idle_for_empty_tab() {
+        let tab = crate::layout::Tab::new(
+            "empty".to_string(),
+            crate::layout::Pane {
+                agent_name: "test".to_string(),
+                vterm: VTerm::new(10, 10),
+                rx: crossbeam_channel::bounded(1).1,
+                id: 1,
+                backend: None,
+                working_dir: None,
+                display_name: None,
+                scroll_offset: 0,
+                has_notification: false,
+                fleet_instance_name: None,
+                last_input_at: None,
+                pending_notification_count: 0,
+                selection: None,
+                source: PaneSource::Local,
+            },
+        );
+        let registry: crate::agent::AgentRegistry =
+            std::sync::Arc::new(parking_lot::Mutex::new(std::collections::HashMap::new()));
+        let result = highest_priority_state(&tab, &registry);
+        assert_eq!(result, AgentState::Idle);
+    }
 }

